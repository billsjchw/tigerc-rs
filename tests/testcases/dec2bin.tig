/* Decimal to binary */
let
    function dec2bin(num: int): string = (
        if num < 2 then (
            num + ""
        ) else (
            dec2bin(num / 2) + (num - num / 2 * 2)
        )
    )
in
    print(0 + " -> " + dec2bin(0) + "\n");
    print(1 + " -> " + dec2bin(1) + "\n");
    print(100 + " -> " + dec2bin(100) + "\n");
    print(9223372036854775807 + " -> " + dec2bin(9223372036854775807) + "\n");
end
