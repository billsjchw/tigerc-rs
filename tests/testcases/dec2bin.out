0 -> 0
1 -> 1
100 -> 1100100
9223372036854775807 -> 111111111111111111111111111111111111111111111111111111111111111
